src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemResponseBuilder.cs

```diff
@@ -81,6 +80,30 @@ public CancelWorkitemResponse BuildCancelResponse()
         return new CancelWorkitemResponse(status, _message);
     }
 
+    /// <inheritdoc />
+    public RetrieveWorkitemResponse BuildRetrieveWorkitemResponse()
+    {
+        var status = WorkitemResponseStatus.Failure;
+
+        if (!_dataset.TryGetSingleValue<ushort>(DicomTag.FailureReason, out var failureReason))
+        {
+            status = WorkitemResponseStatus.Success;
+        }
+        else if (failureReason == FailureReasonCodes.ProcessingFailure ||
```

Comment:
Just want to double check, do we want to send NotFound even on the ProcessingFailure?

Reply:
Yeah same question. Why do we do that?